bin/ignore-extraneous-style-assets.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Webpack plugin to prevent extraneous assets from being emitted for stylesheets.
+ *
+ * JS files and PHP asset files generated by `DependencyExtractionWebpackPlugin` are currently ignored.
+ */
+class IgnoreExtraneousStyleAssets {
+	isCssFile( file ) {
+		return /^(?!!!).+\.(?:sc|sa|c)ss$/.test( file );
+	}
+
+	generateIgnoreRegex( entries ) {
+		const expression = entries
+			.reduce( ( acc, entry ) => `${ acc }|${ entry }.js|${ entry }.asset.php`, '' )
+			.substr( 1 );
```

Comment:
Or to remove any ambiguity we could do something like:

```suggestion
		const expression = entries
			.reduce( ( acc, entry ) => {
				const subExpression = `${ entry }.js|${ entry }.asset.php`;
				return acc === '' ? subExpression : `${ acc }|${ subExpression }`;
			}, '' );
```

Reply:
It feels quite verbose. How about something like:

```suggestion
		const expression = entries
			.reduce( ( acc, entry ) => [ ...acc, `${ entry }.js`, `${ entry }.asset.php` ], [] )
			.join( '|' );
```

_(I feel like we're slowly bikeshedding here 😂 )_

Reply:
Perfecto!